src/include/homestore/btree/detail/btree_mutate_impl.ipp

```diff
@@ -203,7 +203,12 @@ btree_status_t Btree< K, V >::mutate_write_leaf_node(const BtreeNodePtr& my_node
                 BT_DBG_ASSERT(false, "For non-extent keys range-update should be really update and cannot insert");
                 ret = btree_status_t::not_supported;
             } else {
+                const auto new_val_size{(*req.m_newval).serialized_size()};
                 for (auto idx{start_idx}; idx <= end_idx; ++idx) {
+                    if (my_node->available_size(m_bt_cfg) <= new_val_size) {
```

Comment:
I don't quite follow what is this trying to achieve, but result of this line: 208 doesn't seem to rely on the for loop. 
If it is false, it is always false, if it is true, it always do with the 1st idx. 

Also shouldn't the has_more be set to ret and fall through to below logic to handle it? I don't see any caller is handling has_more.

Reply:
1- `update()` changes the value of available_size. Let's assume the current val for entry _`idx`_  has size 1 and the the requested val size = 500. Therefore, it needs to check the requested val size is available or not. (There is a minor bug that I have fixed it to check  extra needed size which I will update this PR). 

2- For the second point this function should return `has_more` then in `put()` it checks the return value and when it is `has_more`, it calls `do_put()` again and in the next call of `do_put()` , `split_nodes()` will occur and the rest of request will perfom.